### PR TITLE
treewide: rename "withX" option to "x11Support"

### PR DESCRIPTION
**Pull request by KAction**
Some packages used "withX" option to enable or disable support for x11, but
most were using x11Support. This commit makes them all uniform.

This is no-rebuild commit.

### CODE REVIEW

File: pkgs/applications/editors/emacs/generic.nix [hunk 1]
```diff
@@ -15,7 +15,7 @@
 , libgccjit, targetPlatform, makeWrapper # native-comp params
 , fetchFromSavannah
 , systemd ? null
-, withX ? !stdenv.isDarwin
+, x11Support ? !stdenv.isDarwin
```
Comment (adisbladis) [inline pkgs/applications/editors/emacs/generic.nix:18]: This makes the x11 option inconsistent with all other options in the derivation.
I don't really like it.

Comment (KAction) [reply pkgs/applications/editors/emacs/generic.nix:18]: I can rename them all. E.g `withMotif` -> `motifSupport`. You okay with that?

Comment (KAction) [reply pkgs/applications/editors/emacs/generic.nix:18]: @adisbladis ping?

**Comment by doronbehar** [comment]
I think this change should not be treewide. Especially since this should be accepted by package maintainers. I'll close this in the meantime since there are also merge conflicts.

**Comment by KAction** [comment]
@doronbehar You suggest that I make couple dozens of single-line pull requests instead?

**Comment by doronbehar** [comment]
Yes.

**Pull request closed without merging**